### ISSUE TITLE
Add destroy compliment to generator commands

### ISSUE THREAD

**Issue by drujensen** [amber-crystal/amber#188]
`amber g controller Post` will create `src/controllers/post.cr`
`amber d controller Post` should delete `src/controllers/post.cr`

Do the same for scaffolding and the rest of the generators.

**Comment by eliasjpr** [comment]
Was this already address?

**Comment by elorest** [comment]
No but @drujensen believes and mostly convinced me that we'll be ok with
out it for now as long as we commit to git before generating things.

Isaac Sloan - isaacsloan.com
Artisan Rails Architect
1391 S Slate Canyon Dr.
Provo, Ut 84606
Cell: (435)-669-5005

**Comment by marksiemers** [comment]
I think this could and should be done in teeplate. I believe the relevant starting points are:
here: https://github.com/mosop/teeplate/blob/master/src/lib/renderer.cr#L47
and here: https://github.com/mosop/teeplate/blob/master/src/lib/rendering_entry.cr#L57

I've opened an issue here: https://github.com/mosop/teeplate/issues/8

**Comment by elorest** [comment]
We decided a couple months ago that this isn't a priority. Seems like I was
the only one who actually uses the destroy option in the rails

**Comment by marksiemers** [comment]
Not a priority, but still a good feature. I love the destroy functionality.

One typo with a scaffold and you're spending 10 minutes deleting files (or worse, NOT deleting them at all).

I usually run a scaffold about 3 times before I have it right.

If the Teeplate maintainer puts it in, then it should be an easy add for us at that point.

**Comment by marksiemers** [comment]
@amberframework/core-team - There is a fix for this in teeplate, but mosop isn't maintaining it anymore.

If we switch to the amberframework fork of this, we can incorporate the work that @aaronstillwell has done.

**Comment by eliasjpr** [comment]
Whats the status of this? Has this been resolved by pointing to Amber Fork?

**Comment by marksiemers** [comment]
@eliasjpr - I think amber will require a change to get this functionality complete in amber.

For files like:
```
...
07:30:59 Generate   | (INFO) skipped   spec/models/spec_helper.cr
...
07:30:59 Generate   | (INFO) skipped   spec/controllers/spec_helper.cr
...
```

There needs to be logic that does not destroy those files (in this example for destroying a scaffold).

**Comment by eliasjpr** [comment]
Agree, I think we can skip those files, after all there are just there to shorten the require. I don't think that we should delete or modify these files contents

**Comment by damianham** [comment]
when using the scaffold generator we get new files but also some files are modified.  In order to support a delete option we really need to roll back to previous versions of any files that were modified during the generate stage.

**Comment by faustinoaq** [comment]
Some update on this?, looks like https://github.com/amberframework/teeplate/pull/2 is already accepted, not merged, though.

**Comment by drujensen** [comment]
One option is to leverage git and rollback to the previous sha.
